site/src/pages/tutorials/tutorial-blog-rest-update.mdx

```diff
@@ -0,0 +1,229 @@
+
+```java filename=BlogService.java
+  @Put("/blogs/:id")
+  public void updateBlogPost(@Param int id, @RequestObject BlogPost blogPost) {
+    BlogPost oldBlogPost = blogPosts.get(id);
+    // Check if the given blog post exists
+    if (oldBlogPost == null) {
+      // Return a Not Found error. See the next section for instructions
+    }
+    BlogPost newBlogPost = new BlogPost(id, blogPost.getTitle(),
+                                            blogPost.getContent(),
+                                            oldBlogPost.getCreatedAt(),
+                                            blogPost.getCreatedAt());
+    blogPosts.put(id, newBlogPost); // Update the info in the map
+    ...
+  }
+```
+
+## 4. Return response
+
+Two possibilities are available for the response. If the blog post to update exists, we respond to the client that the update has been successful. The other response informs the client that the blog post doesn't exist.
+
+### Return error
+
+Let's return an error for the request to update a blog post that doesn't exist.
+
+1. Replace the return type of the `updateBlogPost()` method from `void` to `HttpResponse`.
+2. Return a response using Armeria's <type://HttpResponse> containing `HttpStatus.NOT_FOUND`.
+
+```java filename=BlogService.java highlight=7,10
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+
+public final class BlogService {
+  ...
+  @Put("/blogs/:id")
+  public HttpResponse updateBlogPost(@Param int id, @RequestObject BlogPost blogPost) {
+    ...
+    if (oldBlogPost == null) {
+      return HttpResponse.of(HttpStatus.NOT_FOUND);
+    }
+    ...
+  }
+}
+```
+
+### Return success
+
+Let's return the information updated when the target blog post exists:
+
+1. Replace the return type of the `updateBlogPost()` method from `void` to `HttpResponse`.
+2. Return a response using Armeria's <type://HttpResponse> containing the updated information of the post.
+
+```java filename=BlogService.java highlight=7,9
+import com.linecorp.armeria.common.HttpResponse;
+
+public final class BlogService {
+  ...
+
+  @Put("/blogs/:id")
+  public HttpResponse updateBlogPost(@Param int id, @RequestObject BlogPost blogPost) {
+    ...
+    return HttpResponse.ofJson(newBlogPost);
+  }
+}
+```
+
+## 5. Test updating a blog post
+
+Let's test updating a blog post.
+
+1. Run the server like we did in [Step 1. Create a server](/tutorials/tutorial-blog-rest-create-server) by running the `main()` method or using Gradle. When you see the message, "Server has been started", you can try testing service methods.
+2. Create a couple of blog posts to test updating a blog post and get the ID value returned. Enter the cURL commands below.
+
+  ```bash
+  $ curl --request POST 'localhost:8080/blogs' \
+  -H 'Content-Type: application/json' \
+  -d '{"title":"Test blog for retrieving", "content":"Create to read."}'
```

Comment:
nit: updating and update?

Reply:
Changed the whole example text for READ, UPDATE, and DELETE (62784099df81ece9305254ec75ed62bdc35e480f)